app.ts

```diff
@@ -0,0 +1,252 @@
+		if (this.lastNumber >= this.recordCount) {
+			this.firstNumber = this.recordCount - (rowsPerPage - 1);
+			this.lastNumber = this.recordCount;
+		}
+	}
+
+	updateArrowVisibility() {
+		if (this.firstNumber === 0) {
+			$('.arrow-left').hide();
+		} else {
+			$('.arrow-left').show();
+		}
+		if (this.lastNumber >= this.recordCount) {
+			$('.arrow-right').hide();
+		} else {
+			$('.arrow-right').show();
+		}
+	}
+
+	fetchAndDisplayRecords(): Promise<void> {
+		return dataManager.fetchRecords(this.firstNumber, this.lastNumber)
+			.then(records => {
+				const inputValue = <string>($('#searchInput').val());
+				$("#tableBody").empty();
+				for (const record of records) {
+					// creates row for each record
+					$("tbody").append(`<tr class="row"></tr>`);
+					const lastRow = $(".row:last");
+					for (const value of record) {
+						// assign each record to their column in a specified row
+						lastRow.append(`<td>${value}</td>`);
+						if (value === inputValue) {
+							// highlights the searched row 
+							lastRow.css('background-color', '#DDC0B4');
+						}
+					}
+					$("tbody").append(lastRow);
+				}
+				$('#page').empty().append(`Showing record: ${this.firstNumber} - ${this.lastNumber}`);
+				$('#loader').hide();
+			})
+			.catch(err => {
+				alert('Error while displaying records, reload the page');
+				throw err;
+			});
+	}
+
+	/** recalculates the record range that includes inputValue fromm user */
+	searchRecordsAndResize() {
+		let inputValue = <number>($('#searchInput').val());
+		if (inputValue < 0 || inputValue > this.recordCount) {
+			$('.modal').css('display', 'block');
+			$('.content').append(`<p>${inputValue} is not a number within the range. Please try a different number</p>`);
+			$('#page').empty().append(`Showing record: ${this.firstNumber} - ${this.lastNumber}`);
+			$('#searchInput').val('');
+			return;
+		}
+		let calculatedRows = this.calculatingRows();
+		// divides the calculated max rows in half
+		const halfRange = Math.floor(calculatedRows / 2);
+		this.firstNumber = Math.max(0, inputValue - halfRange);
+		this.lastNumber = Math.min(this.recordCount, this.firstNumber + (calculatedRows - 1));
+	}
+
+	/** Navigates to the next set of records */
+	rightArrow() {
+		$('#searchInput').val('');
+		// retrieves the last row
+		const lastRow = document.querySelector("#recordsTable tbody .row:last-child");
+		// checks if the last row exists 
+		if (lastRow) {
+			const cells = lastRow.querySelectorAll("td");
+			if (cells.length > 0) {
+				// Get the value of the first cell 
+				const lastID = parseFloat(cells[0].textContent || "");
+				// checks if the last value is within range 
+				if (0 <= lastID && lastID <= this.recordCount) {
+					// calculates the first number of the page 
+					this.firstNumber = lastID + 1;
+					const calculatedRows = this.calculatingRows();
+					// calculates the last number of the page 
+					this.lastNumber = this.firstNumber + (calculatedRows - 1);
+				}
```

Comment:
The way I understand this function, you only need this part of the logic. You don't have to go and find the last child. You know what it is because you set it in `this.lastNumber`, right? Replace `lastID` with `this.lastNumber` and then you can remove the rest and just call `this.updateAndDisplayRecords` afterward.